inc/ThirdParty/Plugins/I18n/Weglot.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Plugins\I18n;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Subscriber for compatibility with Weglot.
+ */
+class Weglot implements Subscriber_Interface {
+
+	/**
+	 *  Returns an array of events that this subscriber wants to listen to.
+	 *
+	 * @return array|string[]
+	 */
+	public static function get_subscribed_events() {
+		if ( ! class_exists( 'Context_Weglot' ) ) {
+			return [];
+		}
+
+		return [
+			'rocket_admin_bar_referer' => 'add_langs_to_referer',
+		];
+	}
+
+	/**
+	 * Modify the referer URL by appending the current language from Weglot as a prefix to the URL path.
+	 *
+	 * @param string $referer The original referer URL.
+	 * @return string The modified referer URL with the language as a prefix.
+	 */
+	public function add_langs_to_referer( $referer ) {
+		$current_language = weglot_get_current_language();
```

Comment:
We should add a `function_exists()` check here, in case this function is not loaded for some reason or it's removed later, that would avoid any unexpected error.

Reply:
True, I've added it